File: backend/app/models/item.py
from datetime import datetime
from uuid import UUID
from pydantic import BaseModel, ConfigDict


class ItemResponse(BaseModel):
    model_config = ConfigDict(frozen=True)

    id: UUID
    source_id: UUID
    source_name: str
    title: str
    url: str
    excerpt: str | None = None
    summary: str | None = None
    published_at: datetime | None = None
    category: str
    is_paywalled: bool = False
    is_read: bool = False
    created_at: datetime


class Meta(BaseModel):
    model_config = ConfigDict(frozen=True)

    total: int
    limit: int
    offset: int


class ItemsListResponse(BaseModel):
    model_config = ConfigDict(frozen=True)

    success: bool
    data: list[ItemResponse]
    error: str | None
    meta: Meta

File: frontend/src/types/item.ts
export type Category = "research" | "code" | "community" | "product" | "media";

export interface FeedItem {
  id: string;
  source_id: string;
  source_name: string;
  title: string;
  url: string;
  excerpt: string;
  summary: string | null;
  published_at: string | null;
  category: Category;
  is_paywalled: boolean;
  is_read: boolean;
  created_at: string;
}

export interface Meta {
  total: number;
  limit: number;
  offset: number;
}

export interface ItemsListResponse {
  success: boolean;
  data: FeedItem[];
  error: string | null;
  meta: Meta;
}
